DATA_PREPARATION.md

```diff
@@ -158,4 +157,56 @@ Below is an example of multitask dataset:
 ]
 ```
 
+For `key_value_pair` dataset, an additional field `schema` is required to define the task and the label format. Below is an example of its dataset information dictionary:
+
+```json
+[
+    {
+        "name": "multi-image-question-answer",
+        "version": 1,
+        "description": "Answer question related to one or more images",
+        "type": "key_value_pair",
+        "format": "coco",
+        "root_folder": "key_value_pair/multi_img_qa_20240723",
+        "schema": {
+            "name": "Multi-image QA schema",
+            "description": "Provide answer and a rationale.",
+            "fieldSchema": {
+                "answer": {
+                    "type": "string",
+                    "description": "answer."  
+                },
+                "rationale": {
+                    "type": "string",
+                    "description" :"rationale of answer"
+                }
+            }    
+        },
+        "train": {
+            "index_path": "train.json",
+            "files_for_local_usage": [
+                "train_images.zip"
+            ]
+        },
+        "val": {
+            "index_path": "val.json",
+            "files_for_local_usage": [
+                "val_images.zip"
+            ]
+        }
+    }
+]
+```
+The `fieldSchema` is a dictionary mapping each field to a dictionary called `FieldSchema`. In above example, there are two fields: `answer` and `rationale`. Below is the definition of `FieldSchema`:
+
+| Property | Type | Example | Required? |
+|----------|----------|----------|----------|
+| type | FieldValueType | JSON type: string, number, integer, boolean, array, object. Complex object type: boundingBox.  | yes |
```

Comment:
what is the format of bounding box?

Reply:
OD dataset calls it "bbox", do we want to re-use the same?
OD dataset supports both ltrb or ltwh (default) as well, in absolute pixels.

reference: https://github.com/microsoft/vision-datasets/blob/58dca7a8adc92ca127ad112dd2860bb3d480ac2f/COCO_DATA_FORMAT.md#bbox-format

Reply:
changed to `bbox` with format ltrb. `boundingBox` is the API schema design, but here we can be consistent with OD dataset.  Regarding format, it would be good to only support ltrb for simplicity.